Supports custom Onion URLs on CLI

conorsch opened the pull request:
Closes #1.

Custom Onion URLs for SD instances can now be passed via CLI, regardless of whether those URLs are already specified inside the hardcoded list inside the repo. Sprinkled in a few dev-env-related files like gitignore and a small Makefile. 

Note there's a breaking API change here, in that `--all` must be passed in order to use the hardcoded list; otherwise the script now fails:

```
$ ./sdstatus 
2018/05/13 18:03:31 No args provided. Pass --all to use hardcoded list
```

You can also use `make test` to query a random URL from the hardcoded list, and pass that into the script. Hardly a "test", truly, but does provide useful feedback during development.

conorsch committed:
Adds "--all" bool flag

Trying to make the "scan all instances concurrently" logic optional.
The hardcoded flat file will get out of date quickly, so we'll need to
support reading Onion URLs passed on the command line. Rather than rip
out the hardcoded list support immediately, let's fall back to it.

Simplifies main entrypoint somewhat. Given that we already have a
default boolean flag for the CSV format via the CLI parser, we don't
need an explicit if/else statement.

conorsch committed:
Supports Onion URLs passed as args

Still falls back to the hardcoded list, but now supports items
on the CLI.

conorsch committed:
Adds "sdstatus" binary to gitignore

Shouldn't ever be committed, so let's omit it from the version control.

conorsch committed:
Adds barebones Makefile

Simple "build" target to compile the binary, plus a "test" target which
isn't really a test at all, but does provide some feedback on script
function.

conorsch committed:
Makes --all flag mandatory to use hardcoded list

If `--all` is not passed, then an Onion URL (or several) must be
provided on the CLI. If neither are provided, then the script errors
out. This is a breaking CLI API change, but makes the script more unixy
in its behavior.

kushaldas reviewed (changes requested):
This works, but, breaks the `--csv` flag.

```
$ ./sdstatus v2d6nf2nsvvbgaqe.onion --csv
[
	{
		"Info": {
			"sd_version": "",
			"gpg_fpr": ""
		},
		"Url": "--csv",
		"Available": false
	},
	{
		"Info": {
			"sd_version": "0.7.0",
			"gpg_fpr": "706FD05276BF321C7ED14A0AE7461C79A3839F14"
		},
		"Url": "v2d6nf2nsvvbgaqe.onion",
		"Available": true
	}
]
```

conorsch committed:
Adds --format flag, replaces --csv and --json

More straightforward to declare the format explicitly. Defaults to JSON,
since that makes it easy to integrate with other tools, without having
to parse the CSV structure.

Trims down the "test" target (which is hardly a test), running the
source file directly, rather than building.

conorsch commented:
@kushaldas Made some more changes. There's now a `--format` flag, which accepts `--format json` or `--format csv`, with JSON being the default. Additionally, I removed the logic for reading the local txt file of addresses, since that's highly dependent on running from the git repo.

I realize there are significant API changes, but hopefully as it's still early days, we can shove things around more energetically than usual. Let me know if you have any comments. 

```
 $ sort -r sdonion.txt | head -n 1 | xargs go run main.go --format csv
znig4bc5rlwyj4mz.onion,0.7.0,EDF941416315D6A7EEE414DD1E8A42CB1D386F60
 $ sort -r sdonion.txt | head -n 1 | xargs go run main.go --format json
[
        {
                "Info": {
                        "sd_version": "0.7.0",
                        "gpg_fpr": "EDF941416315D6A7EEE414DD1E8A42CB1D386F60"
                },
                "Url": "znig4bc5rlwyj4mz.onion",
                "Available": true
        }
]
```
